Update release notes for 1.6.0 (#5604)
Cray-HPE/docs-csm: RELEASE_NOTES.md

# Cray System Management (CSM) - Release Notes

[CSM](glossary.md#cray-system-management-csm) 1.6 contains many changes spanning bug fixes, new feature development, and
documentation improvements. This page lists some of the highlights.

## Features

* iSCSI
    * See [iSCSI SBPS](operations/iscsi_sbps/iscsi_sbps.md) for
      details on iSCSI based boot content projection for `rootfs` and `PE` images.
* Multi-Tenancy

> * Bond HSN MAC VLAN interfaces support Slingshot resiliency

Reviewer: MAC VLAN should be `macvlan` as it's a specific virtual interface type.

# Cray System Management (CSM) - Release Notes

[CSM](glossary.md#cray-system-management-csm) 1.6 contains many changes spanning bug fixes, new feature development, and
documentation improvements. This page lists some of the highlights.

## Features

* iSCSI (CASM-4525)
* Multi-Tenancy (CASM-4437)
* Bond HSN MacVLAN interfaces support Slingshot resiliency (CASM-3479)
* HPE Cray OS 24.2
* SLES 15 SP6
* Ceph 18
* SHS 11.1
* CSM can now be upgraded with IUF. See [Upgrade CSM](upgrade/README.md) for more details.
* The [BOS](glossary.md#boot-orchestration-service-bos) API now enforces limits that previously had
  only been recommended. When updating to CSM 1.6, BOS data is migrated to be in compliance with the
  API specification. See [BOS data notice](upgrade/README.md#bos-data-notice) for more details.

> * Please see [iSCSI SBPS](https://github.com/Cray-HPE/docs-csm/blob/release/1.6/operations/iscsi_sbps/iscsi_sbps.md) for

Reviewer 2: ```suggestion
* See [iSCSI SBPS](operations/iscsi_sbps/iscsi_sbps.md) for
```

  details on iSCSI based boot content projection for `rootfs` and `PE` images.
    * The [System Admin Toolkit (SAT)](glossary.md#system-admin-toolkit-sat) is fully included in CSM.
      There is no longer a separate SAT product stream to install. SAT 2.6 releases, which accompanied CSM 1.5, are
      the last releases of SAT as a separate product. For more information,
      see [SAT in CSM](operations/system_admin_toolkit/about_sat/SAT_in_CSM.md).

>     * v1 of Power Control Service (PCS) is active.

Reviewer 2: Are these meant to be sub-bullets of the above iSCSI bullet? They don't seem related to iSCSI.

## New

### New software support

* Resolved CASTS:
    * CAST-26853: "cray bos" doesn't give an error if you accidentally feed it nids and not xnames

>     * CAST-36629: cray-hms-smd-test-functional regex does not match

Reviewer 2: I'm not sure "New software support" is the right heading for the resolved CASTs.
For the BOS one, we added a new BOS v2 option, so I guess it's kind of new software. I am not familiar with the HMS one, but it sounds like a bug fix.

## Improvements

### Automation improvements

* IUF update **TODO**

### Base platform component upgrades

| Platform Component           | Version |
|------------------------------|---------|
| Kubernetes                   | 1.24.17 |
| `containerd`                 | 1.5.16  |
| `istio`                      | 1.19.10 |
| `kiali`                      | 1.75.0  |
| `Strimzi Kafka`              | 0.41.0  |
| `Kyverno`                    | 1.10.7  |
| `keycloak`                   | 21.1.1  |
| `bitnami-etcd` on `ncn-mxxx` | 3.5.0   |
| `bitnami-etcd` for clusters  | 3.5.9   |
| `coredns`                    | 1.8.4   |
| `helm`                       | 3.11.2  |
| `postgresql`                 | 14.8    |
| `postgres-operator`          | 1.8.2   |
| `spire`                      | 0.12.2  |
| `spire-intermediate`         | 1.0.0   |
| `cray-spire`                 | 1.5.5   |
| `metrics-server`             | 0.6.3   |
| `cray-certmanager`           | 1.5.5   |
| `argo-workflows`             | 3.3.6   |
| `argo-workflow-controller`   | 3.4.5   |
| `ceph`                       | 16.2.13 |

### Hardware Mangement System (HMS) improvements

> * `>`75% reduction in OS noise generated by the heartbeat client that runs on compute nodes.

Reviewer 2: ```suggestion
* More than 75% reduction in OS noise generated by the heartbeat client that runs on compute nodes.
```

> * Significant resource utilization improvements in the PCS, FAS, hmcollector, and SMD services which will reduce pod restarts and improve overall functionality.

Reviewer 2: ```suggestion
* Significant resource utilization improvements in the PCS, FAS, hmcollector, and SMD services which will reduce pod restarts and improve overall functionality.

### Scale improvements

* Significant performance improvements in BOS and CFS on large scale systems.
```

* Added support to upgrade CSM through IUF
* Added support for iSCSI based boot content projection for `rootfs` and `PE` images

### New hardware support

* *IMPORTANT* Systems with NVIDIA CPUs and GPUs must see [Known issues](#known-issues)

## Improvements

### Automation improvements

* CSM can now be upgraded with IUF.

### Base platform component upgrades

> | Platform Component           | Version |

Reviewer 2: I did not verify if these versions are actually accurate for CSM 1.6.0. I'm assuming the teams who own these will do that / have done that.

|------------------------------|---------|
| Kubernetes                   | 1.24.17 |
| `containerd`                 | 1.5.16  |
| `istio`                      | 1.19.10 |
| `kiali`                      | 1.75.0  |
| `Strimzi Kafka`              | 0.41.0  |
| `Kyverno`                    | 1.10.7  |
| `keycloak`                   | 21.1.1  |

> | `bitnami-etcd` on `ncn-mxxx` | 3.5.0   |

Reviewer: We don't use bitnami etcd on the master nodes, just plain ol' etcd.

bitnami is just the vendor of the helm chart we use to deploy etcd inside the cluster.

* Updated `Kyverno` documentation
* Added documentation about iSCSI based boot content projection for `rootfs` and `PE` images

> * The SAT documentation moved to be fully included within the [System Admin Toolkit (SAT)](https://cray-hpe.github.io/docs-csm/en-16/operations/system_admin_toolkit/) section of the [CSM Administration Guide](https://cray-hpe.github.io/docs-csm/en-16/operations/).

Reviewer 2: ```suggestion
* The SAT documentation moved to be fully included within the [System Admin Toolkit (SAT)](operations/system_admin_toolkit/README.md) section of the [CSM Administration Guide](operations/README.md).
```

## Removals

> For a list of all features with an announced removal target,

Reviewer 2: BOS v1 is removed in CSM 1.6 -- we do not want to remove this from the release notes.

```suggestion
* [BOS](glossary.md#boot-orchestration-service-bos) v1

For a list of all features with an announced removal target,
```

* [BOS](glossary.md#boot-orchestration-service-bos) v1

For a list of all features with an announced removal target,
see [Removals](introduction/deprecated_features/README.md#removals)

> ## Known issues

Reviewer: I think we should call out the postgres failover issue here.

troubleshooting/known_issues/postgres_database_recovery.md

Neither workaround fully solves the problem, an upgrade of the postgresql components is needed and whenever a master node is rebooted or rebuilt there is a chance that postgresql will failover without the clients noticing resulting in services behaving strangely.

Reviewer: I added this information to the document.

    * A workaround is presented in [IUF does not run the next stage for an activity](troubleshooting/known_issues/iuf_unable_to_run_next_stage.md)

> * iSCSI based boot content projection may fail
>     * If the image to be projected does not have an `etag`

Reviewer 2: ```suggestion
* iSCSI based boot content projection may fail if the image to be projected does not have an `etag`
```

>         * A workaround is presented in [iSCSI SBPS boot failure](https://github.com/Cray-HPE/docs-csm/blob/release/1.6/troubleshooting/known_issues/SBPS_boot_fail.md)

Reviewer 2: I don't think this should be indented over further. Also, the link should be relative.
```suggestion
    * A workaround is presented in [iSCSI SBPS boot failure](troubleshooting/known_issues/SBPS_boot_fail.md)
```

## All Tickets

CASMINST-5476 Evaluate latest zalando postres operator -- determine if we can use upstream without changes
CASMSEC-374 Remove opa-gatekeeper for fresh installs by Jan 17 2023
CASMTRIAGE-4693 vShasta v2: on SLES-15 SP4 NCNs modprobe on the RBD module fails with invalid execution format
CASMTRIAGE-4762 fanta validate certifi is correct version failing
CASMNET-2001 Unbound metrics missing from Prometheus
CASMDIAG-914 Add values for helm charts into customizations.yaml
MTL-2059 Update version of `aaa_base` packages in node images to 10.3.1
MTL-2063 Misc SP4 Updates
CASMHMS-5626 HSM: Re-remove HSM v1 APIs
CASMINST-5767 Update kubernetes / ceph version in file "prerequisites.sh"
CASMTRIAGE-4793 Gamora: Ceph failing to start on S001 after image upgrade

> CASMINST-5799 Replace remaining references to <https://artifactory.algol60.net> in docs for CSM tarball and docs RPM downloads

Reviewer 2: ```suggestion
CASMINST-5799 Replace remaining references to `https://artifactory.algol60.net` in docs for CSM tarball and docs RPM downloads
```

CASMTRIAGE-5704 etcdDatabaseHighFragmentationRatio alert
CASMTRIAGE-5750 Baldar pre-integration - cannot boot any computes
MTL-2209 Default 256M Socket Buffers allow Weave to consume Memory
CASMINST-6555 Add progress indicator to ncn image upload script for usability
CASMTRIAGE-5735 DOCS: Exclude all internal networks from Unbound forwarding
CASMTRIAGE-5734 Docs : Update CSM prerequisite version and CSM release version in docs for CSM-1.5
CASMTRIAGE-5737 "csi pit validate --ceph" suggested remediation reports error for invalid date
CASMTRIAGE-5739 Docs: Update CSM Release value to "1.5.0"
CASMNET-2143 DOCS: Add procedure to persist replica count in Unbound documentation
CASMCMS-8715 SECURITY: Improper Certificate Validation CVE in cfs-operator
CASMCMS-8718 SECURITY: Improper Certificate Validation CVE in cfs
CASMCMS-8722 Use update_external_versions for Python modules in CMS repos
CASMINST-6582 Linting
MTL-2000 Install Packages onto NCNs with cloud-init zypper/packages module
CASMINST-6518 Need to modify the sat command output of iuf
CASMTRIAGE-5846 sysctl issues during prepare-images stage
CASMINST-6608 iuf-cli needs to publish to SP5 or start using NOOS

> CASMPET-6748 boto3 calls to <https://rgw-vip.nmn> fail even when verify cert == false

Reviewer 2: ```suggestion
CASMPET-6748 boto3 calls to `https://rgw-vip.nmn` fail even when verify cert == false
```
